x/merkledb/trieview.go

```diff
@@ -1028,13 +980,9 @@ func (t *trieView) getNodeWithID(id ids.ID, key Path, hasValue bool) (*node, err
 	if err != nil {
 		return nil, err
 	}
-
-	// only need to initialize the id if it's from the parent trie.
-	// nodes in the current view change list have already been initialized.
-	if id != ids.Empty {
-		parentTrieNode.id = id
-	}
-	return parentTrieNode, nil
+	parentTrieNodeClone := parentTrieNode.clone()
+	t.changes.nodes[key] = &change[*node]{before: parentTrieNode, after: parentTrieNodeClone}
```

Comment:
record this here so that we don't have to load the node again from the parentTrie when we call recordNodeChange